### ISSUE TITLE
cloud/docker: update doc for field 'registry'

### ISSUE THREAD

**Issue by elouanKeryell-Even** [ansible/ansible-modules-core#3423]
##### ISSUE TYPE

Documentation Report
##### COMPONENT NAME

docker
##### ANSIBLE VERSION

```
ansible 2.1.0 (devel 22467a0de8) last updated 2016/04/13 11:42:21 (GMT +200)
  lib/ansible/modules/core: (detached HEAD 99cd31140d) last updated 2016/04/13 11:42:31 (GMT +200)
  lib/ansible/modules/extras: (detached HEAD ab2f4c4002) last updated 2016/04/13 11:42:40 (GMT +200)
  config file =
  configured module search path = Default w/o overrides
```
##### SUMMARY

docker module docs describes field `registry` as the "Remote registry URL to pull images from". However, I think this field's only use is for login, not pulling, so the doc is misleading. See the issue I opened on that subject (#3419). It would be nice if the doc could be fixed.

**Comment by stephanarts** [comment]
As commented in: #3419 - I believe the bug is not in the documentation, but in the module itself.

**Comment by elouanKeryell-Even** [comment]
Well either the docs or the code should be updated, so that they are in accordance. I was suggesting to update the doc since it is easier, but yeah I agree with you it would be better to change this 'weird' behaviour in the code. You can open a new issue for code upgrade if you want (if you do I'll close this one).

**Comment by stephanarts** [comment]
I was wrong; You can pull an image from a local registry with the existing docker module.
The way I managed to deploy an image from a local registry server was like this: 

```
- name: create docker container running my-image
  docker:
    name: my-container
    image: docker-registry:5000/my-image:latest
    registry: "https://docker-registry:5000/"
    state: reloaded
    ports:
      - 0.0.0.0:80:80
    pull: always
    restart_policy: always
```

or.

```
- name: create docker container running my-image
  docker:
    name: my-container
    image: docker-registry:5000/my-image:latest
    registry: "http://docker-registry:5000/"
    insecure_registry: true
    state: reloaded
    ports:
      - 0.0.0.0:80:80
    pull: always
    restart_policy: always
```

Which requires `--insecure-registry docker-registry:5000` to be added to the docker-daemon command-line arguments

**Comment by elouanKeryell-Even** [comment]
Yeah that's how I do too

**Comment by ansibot** [comment]
@olsaki, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@cove, @softzilla, @zfil, @ThomasSteinbach, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@cove, @softzilla, @zfil, @ThomasSteinbach, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.
